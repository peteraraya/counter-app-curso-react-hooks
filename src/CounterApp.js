// Proptypes
import propTypes from 'prop-types';
import { useState } from 'react';

function CounterApp({value=10}) {

  //hooks
  const [counter, setCounter] = useState(value);

  const  handleAdd = (e) => {
   // utilizamos el useState
    setCounter(counter + 1);

    //setCounter( (c) => c + 1); --> Otra opción que se puede utilizar en algun otro caso
    // return () => console.log('Hola mundo') // cuando utilizo () en la llamada;
  }
  const handleReset = () => setCounter(value);

  const handleSubstract = () => setCounter(counter - 1);


  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>

      <button
        className="btn btn-primary me-2"
        onClick={handleAdd} >
          +1
      </button>
      <button
        className="btn btn-success me-2"
        onClick={handleReset} >
          Reset
      </button>
      <button
        className="btn btn-danger me-2"
        onClick={handleSubstract} >
          -1
      </button>
    </>
  );
}

CounterApp.prototypes = {
  value: propTypes.number.isRequired
}

export default CounterApp;

// Toda función en js que no tiene un return explito retorna undefined
// buena practica escribir codigo facil de leer
