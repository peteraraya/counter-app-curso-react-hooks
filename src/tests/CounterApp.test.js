
import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Pruebas en <CounterApp />', () => {
  // lo asigno para no perder el intellinse
  let wrapper = shallow(<CounterApp />);;

  // Ciclo de vida de las pruebas
  beforeEach(() => {
    /** se ejecuta antes de de cada una de las evaluaciones */
    wrapper = shallow(<CounterApp />);
  });

  test('debe de mostrar <CounterApp /> correctamente', () => { 
    // definimos nuestro wrapper de forma global
    expect(wrapper).toMatchSnapshot();
    
  });

  test('debe de mostrar el valor por defecto de 100', () => {

    const wrapper = shallow(<CounterApp value={100} />);

    const counterText = wrapper.find('h2').text().trim();

    expect(counterText).toBe('100');

  });


  test('debe incrementar con el botón +1', () => {
   
    // simulate
    wrapper.find('button').at(0).simulate('click');

    const counterText = wrapper.find('h2').text().trim();

    expect(counterText).toBe('11');

  });

  test('debe decrementar con el botón -1', () => {
   
    // simulate
    wrapper.find('button').at(2).simulate('click');

    const counterText = wrapper.find('h2').text().trim();

    expect(counterText).toBe('9');

  });
  
  test('debe de colocar el valor por defecto con el botón reset', () => {

    const wrapper = shallow(<CounterApp value={105} />);

    // necesito 2 referencias
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(0).simulate('click');
    
    wrapper.find('button').at(1).simulate('click');
    const counterText = wrapper.find('h2').text().trim();
    
    expect(counterText).toBe('105');
    
  });
  
  
  
});
