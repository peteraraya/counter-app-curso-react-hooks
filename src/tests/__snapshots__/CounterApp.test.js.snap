// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe de mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    className="btn btn-primary me-2"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="btn btn-success me-2"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    className="btn btn-danger me-2"
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
